app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -227,10 +227,14 @@ def associate
       api :PUT, "/compute_resources/:id/refresh_cache/", N_("Refresh Compute Resource Cache")
       param :id, :identifier, :required => true
       def refresh_cache
-        if @compute_resource.respond_to?(:refresh_cache) && @compute_resource.refresh_cache
-          render_message(_('Successfully refreshed the cache.'))
+        if @compute_resource.respond_to?(:refresh_cache)
+          if @compute_resource.refresh_cache
+            render_message(_('Successfully refreshed the cache.'))
+          else
+            render_message(_('Failed to refresh the cache.'), :status => :unprocessable_entity)
+          end
         else
-          render_message(_('Failed to refresh the cache.'), :status => :unprocessable_entity)
+          raise ::Foreman::Exception.new(N_("Cache refreshing is not supported for %s") % @compute_resource.provider_friendly_name)
```

Comment:
Rails noob question: is the `N_()` here needed or should it be `_()` like in the UI controller?

Reply:
https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions - tl;dr - exceptions need the untranslated string to generate error code correctly, the string can be translated later when displayed.

Reply:
Actually looking at that, this line should be:
```suggestion
          raise ::Foreman::Exception.new(N_("Cache refreshing is not supported for %s"), @compute_resource.provider_friendly_name)
```

Reply:
See, asking questions in reviews because you don't know is always a good idea. I learned something and we found an issue :)